### ISSUE TITLE
Adguard uses incorrect names

### ISSUE THREAD

**Issue by Mister-XY** [AdguardTeam/AdGuardHome#7532]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, ARMv7

### Installation

Other (please mention in the description)

### Setup

On one machine

### AdGuard Home version

0.107.55

### Action

Replace the following command with the one you're calling or a
description of the failing action:

```sh
 nslookup 192.168.178.19
19.178.168.192.in-addr.arpa     name = Thomas-Smartphone-Xiaomi.fritz.box.
19.178.168.192.in-addr.arpa     name = Xiaomi-11-Lite-5G-NE.fritz.box.

```
![Screenshot 2024-12-27 095319](https://github.com/user-attachments/assets/26d844b3-2942-4f4a-b0bc-e2895da83f95)

### Expected result

I want see the correct Names how it set like Fritzbox Router
![Screenshot 2024-12-27 092548](https://github.com/user-attachments/assets/6540f837-d754-45e9-82fa-7e593b49f3a8)

### Actual result

I see the old names or what they were called before the Fritzbox was renamed.
![Screenshot 2024-12-27 094024](https://github.com/user-attachments/assets/ec97208f-c460-4b8a-9b1e-3365cbf14df3)

### Additional information and/or screenshots

For a long time now, I've noticed that all the clients in Adguard Home are named incorrectly. This means that they often get the name they had before they were renamed.
Of course, you could store all the names in /etc/host with the corresponding IP, but this solution is not very well thought out. If I change the IP in the router, then it also has to be changed manually in /etc/host. This means that you are basically always forced to do something else. This problem does not exist in Pihole.

**Comment by Mister-XY** [comment]
Yes, I changed the names a long time ago. I didn't have Adguard on the PI back then.

**Comment by whyisthisbroken** [comment]
The rDNS Feature in AGH is broken since, i dont know...ever ?
I tried also with my got no luck, so i added all names in the hosts file. (/etc/hosts)

Now it works.

**Comment by Mister-XY** [comment]
I think the names were correct 1-2 years ago and after an update from Adguard they were no longer correct and are displayed as they are now. I'm ruling out the /etc/host thing because the name changes when the IP changes.
Suppose I enter the name "Hummel" for 192.168.178.9
and for some reason I have to assign the IP 192.168.178.9 differently in the router, then the name for the IP is no longer correct.
That means I would have to remember to change the /etc/host as well.

**Comment by saint-lascivious** [comment]
> The rDNS Feature in AGH is broken since, i dont know...ever ?

I'm just chiming in here to say that rDNS works fine here for local domain discovery (I have some magic to map out my local domain in unbound local-data local-data-ptr config).

My first question would be asking whether or not PTR records for your local domain actually exist/are provided by anything in your network, and my second question would be if yes is AGH actually configured to direct reverse lookups there?

**Comment by whyisthisbroken** [comment]
Can you share your magic with me you created in Unbound? I've tried it before, but it's not working properly

**Comment by whyisthisbroken** [comment]
Can you share your magic with me you created in Unbound? I've tried it before, but it's not working properly

**Comment by saint-lascivious** [comment]
> Can you share your magic with me you created in Unbound?

Sure.

Well. I can't really share the magic to generate the config for a given host as that's all very proprietary to my local network and high availability stuff, I can share an example of the resulting config covering a range of common values (`A`, `AAAA`, `TXT` and `PTR`), additionally demonstrating two slightly different ways of handling `PTR` records with either traditional `local-data` or `local-data-ptr` shorthand.

The example uses an FQDN (`sub.domain.tld`), but it could just as easily be a domain or hostname supplied instead.

Note that this configuration is found under the `server:` configuration header.

`server:
    # sub.domain.tld
    local-zone: "sub.domain.tld." static
    local-data: "sub.domain.tld. 3600 IN A 192.168.1.10"
    local-data: "sub.domain.tld. 3600 IN AAAA fd80::aaaa:bbbb:cccc:1234"
    local-data: "sub.domain.tld. 3600 IN AAAA fe80::aaaa:bbbb:cccc:1234"
    local-data: "sub.domain.tld. 3600 IN AAAA 2404:4404::aaaa:bbbb:cccc:1234"
    local-data: 'sub.domain.tld. 3600 TXT "sub.domain.tld"'
    local-data: "10.1.168.192.in-addr.arpa. 10800 IN PTR sub.domain.tld."
    local-data-ptr: "fd80::aaaa:bbbb:cccc:1234 3600 sub.domain.tld"
    local-data-ptr: "fe80::aaaa:bbbb:cccc:1234 3600 sub.domain.tld"
    local-data-ptr: "2404:4404::aaaa:bbbb:cccc:1234 3600 sub.domain.tld"
`

**Comment by gpz1100** [comment]
I was having a some what similar issue.

For example, I have 2 entries defined in custom filters.

```
21.1.10.10.in-addr.arpa^$dnsrewrite=NOERROR;PTR;host1.local.domain
 1.1.10.10.in-addr.arpa^$dnsrewrite=NOERROR;PTR;host2.domain.com
```

When I do a nslookup for 10.10.1.21 it would show both. The fix was beyond simple. The reason this happens is because 10.10.1.1 is a subset of 10.10.1.21. So it was matching both.  To fix, use || for both or at least for the 1.1.10.10 entry.  || means match from the beginning, not anywhere.  RTFM.... always.. 👍

**Comment by Mister-XY** [comment]
Is the problem solved or a solution for the problem?

**Comment by myssv** [comment]
One year later same problem here:

<img width="1166" height="110" alt="Image" src="https://github.com/user-attachments/assets/da121dc1-0d6c-4856-8ad6-9e997ce69b8e" />

Doing a nslookup to AdGuardHome and my Fritz!Box show the correct name

<img width="562" height="263" alt="Image" src="https://github.com/user-attachments/assets/9377c992-8d8d-4fa9-8682-dde350b9039c" />

I deleted the record in the Fritz!Box and let it add again vie DHCP with correct DNS names

<img width="758" height="309" alt="Image" src="https://github.com/user-attachments/assets/df6a00fa-46ec-451d-9b9b-4fdf47f5873c" />

Why do this happen?
